Added tests and README.MD

KTSLX opened the pull request.

KTSLX committed:
Added tests and README.MD

AlexKitIt commented on locators.py:
```diff
@@ -0,0 +1,46 @@
+
+Логин
+//*[@id="root"]/div/main/div/form/fieldset[1]/div/div/input[@type="text"]  #поле ввода email на странице входа
```
Нужно исправить: здесь и далее: все локаторы хранятся тут в переменных, чтобы их можно было переиспользвать в проекте а не просто так
Кроме того, здесь и далее: использование порядкового номера элемента в dom является весьма нестабильным способом поиска, необходимо привязываться к атрибутам конкретного элемента
Кроме того ходить от root и использовать абсолютные пути нельзя - очень ненадежно, добавится на пути новый элемент - тест ляжет
Все локаторы необходимо переписать

AlexKitIt commented on login_module.py:
```diff
@@ -0,0 +1,15 @@
+
+def login(driver):
+    # Ожидание появления кнопки "Войти" под формой логина
+    WebDriverWait(driver, 3).until(EC.visibility_of_element_located((By.XPATH, '//*[@id="root"]/div/main/div/form/button[text()="Войти"]')))
```
Нужно исправить: здесь и далее: все локаторы хранятся отдельно от тестов в модуле locators

AlexKitIt commented on login_module.py:
```diff
@@ -0,0 +1,15 @@
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+def login(driver):
```
Нужно исправить: судя по функционалу это должно быть фикстурой и жить в conftest

AlexKitIt commented on tests/conftest.py:
```diff
@@ -0,0 +1,15 @@
+def driver():
+    driver = webdriver.Chrome()
+    driver.get("https://stellarburgers.nomoreparties.site/")
+    WebDriverWait(driver, 3).until(EC.visibility_of_element_located(
+        (By.XPATH, '//*[@id="root"]/div/main/section[2]/div/button[text()="Войти в аккаунт"]')))
```
Нужно исправить: инстанс драйвера должен подниматься и закрываться в этой фикстуре, ничего другого в ней быть не должно. Если нужно еще что-то - пишется отдельная фикстура

AlexKitIt commented on tests/test_constructor_navigation.py:
```diff
@@ -0,0 +1,32 @@
+from login_module import login
+
+
+def test_constructor_navigation_scrolls(driver):
```
Нужно исправить: здесь и далее по тестам: тестовые методы в модулях объединяются в тестовые классы

AlexKitIt replied:
Нужно исправить: тестов на переход по табам должно быть три изолированных, по одному на каждый таб

AlexKitIt commented on tests/test_constructor_navigation.py:
```diff
@@ -0,0 +1,32 @@
+    # Нажатие на кнопку "Булки"
+    driver.find_element(By.XPATH, '//*[@id="root"]/div/main/section[1]/div[1]/div[1]').click()
+    # Ожидание скролла до булок
+    WebDriverWait(driver, 3).until(
+        EC.visibility_of_element_located((By.XPATH, '//*[@id="root"]/div/main/section[1]/div[2]/h2[1]')))
```
Нужно исправить: здесь и далее: тест всегда заканчивается assert где происходит сравнение ожидаемого результата с фактическим. Это не тест

AlexKitIt commented on tests/test_login.py:
```diff
@@ -0,0 +1,47 @@
+    ]
+)
+def test_login_different_start_points_logged_in(driver, test_type, login_button, login_start_page):
+    if test_type == 'main_page':
```
Нужно исправить: условий в тестах быть не может, тест всегда однозначен

KTSLX replied:
test_type берётся из кортежа в параметризации. нужен для определения начальной точки процесса логина

AlexKitIt commented on tests/test_login.py:
```diff
@@ -0,0 +1,47 @@
+        EC.visibility_of_element_located((By.XPATH, '//*[@id="root"]/div/main/section[1]/h1[text()="Соберите бургер"]'))
+    )
+    assert main_page_header is not None, "Login failed: 'Соберите бургер' header not found."
+    print("Login успешен и отображается главная страница.")
```
Нужно исправить: здесь и далее: принтов быть не должно

AlexKitIt commented on login_module.py:
```diff
@@ -0,0 +1,15 @@
+    # Ожидание появления кнопки "Войти" под формой логина
+    WebDriverWait(driver, 3).until(EC.visibility_of_element_located((By.XPATH, '//*[@id="root"]/div/main/div/form/button[text()="Войти"]')))
+    # Ввод логина
+    driver.find_element(By.XPATH, '//*[@id="root"]/div/main/div/form/fieldset[1]/div/div/input[@type="text"]').send_keys('akuts9123@gmail.com')
+    # Ввод пароля
+    driver.find_element(By.XPATH, '//*[@id="root"]/div/main/div/form/fieldset[2]/div/div/input[@type="password"]').send_keys('111111')
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще

KTSLX committed:
Переработал файл locators.py - все локаторы имеют свою переменную, убрал неиспользуемые локаторы.
Изменил все локаторы (отошёл от использования порядковых номеров DOM, перестал использовать абсолютные пути, привязался к конкретным атрибутам элементов)
Модуль login_module.py оставил как есть (не стал переносить в фикстуры), т.к. он используется не во всех тестах.
Убрал из фикстуры driver ненужное ожидание появления элемента
Добавил ассерты во всех тестах где не было.
Тест навигации в конструкторе разбил на 3 отдельных используя параметризацию.
Все тесты используют локаторы из переменных в файле locators.py
Убрал принты из всех тестов
Добавил модуль data_module.py в котором хранятся данные для логина/регистрации
